windrose/windrose_main.py

```diff
@@ -0,0 +1,123 @@
+import os
+import pandas as pd
+import numpy as np
+from windrose import WindroseAnalysis
+import time
+from pathlib import Path
+import argparse
+
+
+class WindroseProcessor:
+    def __init__(self, root_directory, directory='windRoseData'):
+        self.root_directory = Path(root_directory)
+        self.directory = Path(directory)
+
+        if not self.directory.exists():
+            self.directory.mkdir(parents=True)
+
+    def process_site_files(self):
+        site_list = [
+            d for d in self.root_directory.iterdir()
+            if d.is_dir()
+        ]
+        # print(site_list)
+
+        for site in site_list:
+            file_list = os.listdir(site)
```

Comment:
If you are using `pathlib` don't mix with using `os` , be consistent on the approach